common/src/main/java/org/apache/iceberg/common/DynMethods.java

```diff
@@ -161,6 +161,8 @@ private BoundMethod(UnboundMethod method, Object receiver) {
       this.receiver = receiver;
     }
 
+    /** @deprecated since 1.6.0, will be removed in 1.7.0 */
```

Comment:
can you elaborate please why these are being deprecated? It's not obvious for a reviewer

Reply:
The `Dyn*` classs are copied from Parquet and contain more code than needed by Iceberg codebase. I assume these utilities  were not meant to be API of the project and so any dead code here should be removed. Per revapi checks, removal requires prior deprecation.

Reply:
I don't mind having a few additional methods around. They come in a pair of `{invoke, invokeChecked}`. It looks weird if they are sometime defined, and sometimes missing.

Reply:
My thinking is that any variant needed in the future is very easy to add when needed and doesn't necessitate keeping current-dead-code around. Removing dead code clears the picture and open eyes on further simplifications.  Does this make sense? BTW recently, i was starring at a pretty complex class that was negatively impacted by some other changes only to eventually realize it's used only in its own tests and can simply be deleted. What a relief.

Reply:
I still think we should just remove the dead code. But I am also ok leaving it around if it helps something. let me know which direction you want this to go

Reply:
I like to either remove stuff or keep it there to keep the full picture. If we decide to remove this, I think we can clean it up even further. As part of https://github.com/apache/iceberg/pull/10542/ we can mark `invokeChecked` as deprecated there. Then we can just inline `invokeChecked` in `DynMethods`:

https://github.com/apache/iceberg/blob/91fbcaa62c25308aa815557dd2c0041f75530705/common/src/main/java/org/apache/iceberg/common/DynMethods.java#L54-L68

Less is more!

Reply:
i'm not sure i follow. for now i just dropped deprecation of invokeChecked methods in this PR. My intuitive reaction to IDE warning about dead code such as unused method is to remove it, but clearly it's not as simple. Maybe we do this as a follow-up, if needed.

Reply:
What I'm trying to say is that we should mark [this `invokeChecked`](https://github.com/apache/iceberg/blob/ff7e833c64806915af47499cbecf926ecdc875ee/common/src/main/java/org/apache/iceberg/common/DynMethods.java#L55) as deprecated as well. Since that's only being used from [`invoke`](https://github.com/apache/iceberg/blob/ff7e833c64806915af47499cbecf926ecdc875ee/common/src/main/java/org/apache/iceberg/common/DynMethods.java#L72). This way we can remove `invokeChecked` completely from the public API.

Reply:
you're right. done!